node/pkg/driver/node_utils_test.go

```diff
@@ -101,3 +107,101 @@ func TestParseIscsiInitiators(t *testing.T) {
 	}
 
 }
+
+func TestParseFCPortsName(t *testing.T) {
+	testCases := []struct {
+		name          string
+		file_contents []string
+		err	      error
+		expErr        error
+		expFCPorts    []string
+	}{
+		{
+			name: "fc port file with wrong content",
+			file_contents: []string{"wrong content"},
+			expErr: fmt.Errorf(driver.ErrorWhileTryingToReadFC, "wrong content"),
+		},
+		{
+			name: "fc unsupported",
+			expErr: fmt.Errorf(driver.ErrorUnsupportedConnectivityType, "FC"),
+		},
+		{
+			name: "one FC port",
+			file_contents: []string{"0x10000000c9934d9f"},
+			expFCPorts: []string{"10000000c9934d9f"},
+		},
+		{
+			name: "one FC port file with wrong content, another is good",
+			file_contents: []string{"wrong content", "0x10000000c9934dab"},
+			expFCPorts: []string{"10000000c9934dab"},
+		},
+		{
+			name: "one fc port file with wrong content, aonther file path is inexistent",
+			file_contents: []string{"wrong content", ""},
+			expErr: errors.New("Error while tring to get FC port from string: wrong content.,open /non/existent/path: no such file or directory"),
+		},
+		{
+			name: "two FC ports",
+			file_contents: []string{"0x10000000c9934d9f", "0x10000000c9934dab"},
+			expFCPorts: []string{"10000000c9934d9f", "10000000c9934dab"},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			filePath := ""
+			var fpaths []string
+
+			for _, file_content := range tc.file_contents {
+				if file_content != "" {
+					tmpFile, err := ioutil.TempFile(os.TempDir(), "fc-")
+					fmt.Println(tmpFile)
+					if err != nil {
+						t.Fatalf("Cannot create temporary file : %v", err)
+					}
+
+					defer os.Remove(tmpFile.Name())
+
+					text := []byte(file_content)
+					if _, err = tmpFile.Write(text); err != nil {
+						t.Fatalf("Failed to write to temporary file: %v", err)
+					}
+
+					if err := tmpFile.Close(); err != nil {
+						t.Fatalf(err.Error())
+					}
+					filePath = tmpFile.Name()
+				} else {
+					filePath = "/non/existent/path"
+				}
+
+				fpaths = append(fpaths, filePath)
+			}
+
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+
+			fake_executer := mocks.NewMockExecuterInterface(mockCtrl)
+			devicePath := "/sys/class/fc_host/host*/port_name"
+			fake_executer.EXPECT().FilepathGlob(devicePath).Return(fpaths, tc.err)
+			nodeUtils := driver.NewNodeUtils(fake_executer)
+
+			fcs, err := nodeUtils.ParseFCPortsName(devicePath)
+
+			if tc.expErr != nil {
+				if err.Error() != tc.expErr.Error() {
+					t.Fatalf("Expecting err: expected %v, got %v", tc.expErr, err)
+				}
+
+			} else {
+				if err != nil {
+					t.Fatalf("err is not nil. got: %v", err)
+				}
+				if !reflect.DeepEqual(fcs, tc.expFCPorts) {
+					t.Fatalf("scheme mismatches: expected %v, got %v", tc.expFCPorts, fcs)
+				}
+
+			}
+		})
```

Comment:
for some reason reviewable is not working, please try to fix it so it willl be easier to continue review